lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -159,10 +175,12 @@ def add_card(post, credit_card, options)
           post[:card][:securityCode] = {}
           post[:card][:securityCode][:indicator] = 1
           post[:card][:securityCode][:value] = credit_card.verification_value
+          post[:card][:present] = 'Y'
         else
           post[:card] = {} if post[:card].nil?
           post[:card][:token] = {}
           post[:card][:token][:value] = credit_card
+          post[:card][:present] = 'N'
         end
       end
```

Comment:
I think it would be nice to rename the local variable in this method from `credit_card` to `payment` or `payment_method` because the method is for both credit cards and third party tokens.

Reply:
Updated as well.